servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/LoadBalancerBuilder.java

```diff
@@ -0,0 +1,90 @@
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+import io.servicetalk.client.api.LoadBalancer;
+import io.servicetalk.client.api.LoadBalancerFactory;
+import io.servicetalk.client.api.ServiceDiscoverer;
+import io.servicetalk.concurrent.api.Executor;
+
+import java.time.Duration;
+
+public interface LoadBalancerBuilder<ResolvedAddress, C extends LoadBalancedConnection> {
```

Comment:
do we need an interface for a builder, vs a concrete class? Will we need this interface internally? Often times concrete builders may have different ways to configure themselves and may not needs the same options.

Reply:
Yes, interface is helpful when we add support for ServiceLoader providers

Reply:
@idelpivnitskiy, couldn't we do something like the `RoundRobinLoadBalancers` strategy of making the constructor private and funneling all builders through the providers that get first dibs to set defaults?

Reply:
I pushed a commit that seems along the lines of what @Scottmitch is suggesting. I like it.

Reply:
It's a simplistic approach that we use for cases, when we don't need a ServiceLoader provider, like RedirectConfigBuilder, ProxyConfigBuilder, H1ProtocolConfigBuilder, etc. For major objects, like client/server builder, DnsServiceDiscovererBuilder, LB builder it's extremely useful to have access through providers. Providers require them to be interfaces bcz users need to be able to wrap a builder. Wrapper monitors what users set on the builder explicitly and at the build time can modify or add if users missed something. See simple examples in `HttpProvidersTest`.